### PR TITLE
Upgrade Spark Cluster

### PR DESCRIPTION
**Pull request by amCap1712**
All of these upgrades are already in place on production, updating here for testing and documentation purposes.

1. Upgrade Spark to 3.3.0, Hadoop to 3.3.4, Java to 17 (only for spark, hadoop still runs on Java 11)
2. Replace YARN with Spark Standalone cluster
3. Add systemd service file for request consumer

### CODE REVIEW

File: docker/spark-cluster-config/michael/request_consumer.service [hunk 1]
```diff
@@ -0,0 +1,11 @@
+[Unit]
+Description=The ListenBrainz Request Consumer for Spark
+
+[Service]
+WorkingDirectory=/home/request_consumer/listenbrainz-server/
+ExecStart=/home/request_consumer/listenbrainz-server/docker/start-spark-request-consumer.sh
+User=root
```
Comment (amCap1712) [inline docker/spark-cluster-config/michael/request_consumer.service:7]: I tried using the request_consumer user but ran into troubles with file permissions errors in /data/tmp directories and HDFS. So leaving it as root for now. Also, request_consumer user doesn't exist on workers nodes so will need to figure out how/if it'll work,

Comment (mayhem) [reply docker/spark-cluster-config/michael/request_consumer.service:7]: Agreed, but this is a good start. Maybe leave a TODO comment here?

**Review by mayhem — APPROVED** [review]
easy enough!

File: docker/spark-cluster-config/michael/request_consumer.service (hunk 1 above)
Comment (amCap1712) [reply docker/spark-cluster-config/michael/request_consumer.service:9]: Done.

**Pull request merged**